### ISSUE TITLE
Compatibility with apple silicon

### ISSUE THREAD

**Issue by rob2go** [EddyRivasLab/hmmer#228]
Dr. Eddy

I'd like to ask if a compatibility to Apple Silicon is thought be developed any time soon.
I hope yes, because HMMER is very important for many programs.

Thank you in advance

> configure: Configuring HMMER3 for your system.
checking build system type... Invalid configuration `arm64-apple-darwin20.3.0': machine `arm64-apple' not recognized
configure: error: /bin/sh ./config.sub arm64-apple-darwin20.3.0 failed

**Comment by cryptogenomicon** [comment]
(From previous iss #227:)

To build HMMER on the new Apple Silicon Macs, you need to have Apple's [Rosetta2 package](https://en.wikipedia.org/wiki/Rosetta_(software)#Rosetta_2) installed, which gives you support in Xcode for the Intel->ARM transition. On OS/X Big Sur, Apple does not install Rosetta2 with Xcode by default. To install Rosetta2, see [these instructions](https://derflounder.wordpress.com/2020/11/17/installing-rosetta-2-on-apple-silicon-macs/). Then HMMER should build from source as usual.

**Comment by rob2go** [comment]
Ok thank you. For me it continues to not work.
Since the first day, rosetta was installed as most of my apps are yet intel based.

I used the script proposed because maybe there should be a version for terminal.
But I got the message

> Rosetta is already installed. Nothing to do.

And I also tried to follow the other comment that proposes to check the "open using Rosetta" when we go to "Get info" on Terminal.app.

None solve my problem.
If I download Hmmer direct from your website and  ./configure, I still get the same previous message.
Using homebrew, 

> brew install --build-from-source hmmer

I get the following


```
==> ./configure --prefix=/opt/homebrew/Cellar/hmmer/3.3.2
Last 15 lines from /Users/rob/Library/Logs/Homebrew/hmmer/01.configure:
checking whether ln -s works... yes
checking for a sed that does not truncate output... /opt/homebrew/Library/Homebrew/shims/mac/super/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking whether clang is Clang... yes
checking whether Clang needs flag to prevent "argument unused" warning when linking with -pthread... no
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking whether clang can compile our Altivec/VMX code... no
checking whether clang can compile our SSE code... no
configure: ::::::::::--- no vector instruction set ---::::::::::
configure: HMMER3 requires SSE or VMX vector instructions.
configure: Supported platforms are x86 (Intel/AMD) and PowerPC.
configure: error: No supported vectorization found for your machine.

```
I will then continue to investigate my issue. If it worked with others, maybe there is something going on with my configurations.
Thank you!

**Comment by npcarter** [comment]
Hello,

I'm Nick Carter, a programmer on the HMMER effort. The error messages you're seeing suggest to me that there's a problem with your Rosetta 2 installation or that your system isn't trying to use Rosetta 2 to build HMMER, as they're exactly the error messages you'd get when trying to build HMMER on a M1 Mac without Rosetta 2. Here are a few suggestions to try:

1) The first thing I'd suggest is using the Activity Monitor app to verify that your Terminal window is running under Rosetta 2. If Terminal is running under Rosetta 2, there should be a "zsh" task listed on the Activity Monitor with an architecture of "Intel", while the rest of the running applications (or at least most of them) have an architecture of "Apple". If zsh is running with an architecture of "Apple", it's not running under Rosetta 2, and HMMER won't build correctly.

2) https://stackoverflow.com/questions/64882584/how-to-run-the-homebrew-installer-under-rosetta-2-on-m1-macbook says that you can force compilation of Homebrew packages by using the variant "arch -x86_64 brew install <package>". You may have to use the command mentioned in the thread to install an x86 version of Homebrew first. On our M1 Mac, Homebrew code builds under Rosetta 2 without this flag, but it may help you.

3) If none of those work, you might try downloading another terminal emulator, such as kitty. On my Mac, building HMMER from within that terminal application works without any settings or special commands, probably because the terminal application is x86 code, and so automatically runs under Rosetta.

Could you try these and let us know if they work?

-Nick

https://stackoverflow.com/questions/64882584/how-to-run-the-homebrew-installer-under-rosetta-2-on-m1-macbook	https://stackoverflow.com/questions/64882584/how-to-run-the-homebrew-installer-under-rosetta-2-on-m1-macbook	[How to run the Homebrew installer under Rosetta 2 on M1 Macbook
I'm on the M1 MacBook. This is the error when I try to install Homebrew: /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)" Homebrew is not...](https://stackoverflow.com/questions/64882584/how-to-run-the-homebrew-installer-under-rosetta-2-on-m1-macbook)	https://stackoverflow.com/questions/64882584/how-to-run-the-homebrew-installer-under-rosetta-2-on-m1-macbook

[Nick Carter - Chat @ Spike](https://spikenow.com/r/a/?ref=spike-organic-signature&_ts=xacwk)	[xacwk]

On February 28, 2021 at 17:59 GMT, rob2go <notifications@github.com> wrote:

Ok thank you. For me it continues to not work.
Since the first day, rosetta was installed as most of my apps are yet intel based.

I used the script proposed because maybe there should be a version for terminal.
But I got the message

Rosetta is already installed. Nothing to do.

And I also tried to follow the other comment that proposes to check the "open using Rosetta" when we go to "Get info" on Terminal.app.

None solve my problem.
If I download Hmmer direct from your website and ./configure, I still get the same previous message.
Using homebrew,

brew install --build-from-source hmmer

I get the following
==> ./configure --prefix=/opt/homebrew/Cellar/hmmer/3.3.2 Last 15 lines from /Users/rob/Library/Logs/Homebrew/hmmer/01.configure: checking whether ln -s works... yes checking for a sed that does not truncate output... /opt/homebrew/Library/Homebrew/shims/mac/super/sed checking for grep that handles long lines and -e... /usr/bin/grep checking for egrep... /usr/bin/grep -E checking whether clang is Clang... yes checking whether Clang needs flag to prevent "argument unused" warning when linking with -pthread... no checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE checking whether more special flags are required for pthreads... no checking for PTHREAD_PRIO_INHERIT... yes checking whether clang can compile our Altivec/VMX code... no checking whether clang can compile our SSE code... no configure: ::::::::::--- no vector instruction set ---:::::::::: configure: HMMER3 requires SSE or VMX vector instructions. configure: Supported platforms are x86 (Intel/AMD) and PowerPC. configure: error: No supported vectorization found for your machine.
I will then continue to investigate my issue. If it worked with others, maybe there is something going on with my configurations.
Thank you!

**Comment by rob2go** [comment]
Hello

Thank you for you answer. I will soon try it.
I have already done the arch command before. But I'd like to point out that when this thread on stack overflow started, it was just in the beginning of the M1 and Homebrew was still being translated by rosetta. But now, homebrew was updated and don't need Rosetta anymore and if something still depends on Rosetta's translation, then it is not going to run. 

This is why Nigel Davies propose to install 2 copies of homebrew in different locations, under the 2 different architectures.

I will try one of the solutions and let you know.
Thank you

**Comment by rob2go** [comment]
ok thank you for your help.
I installed kityy. I verified on the Activity monitor that it was effectively running "bash" under intel. Everything fine but even like that, not possible to install hmmer.

So I installed homebrew under /usr/local/bin
And now it works :-)

`arch -x86_64 /usr/local/bin/brew install hmmer`

Like that, does not matter if I install using kitty or the apple terminal. The important thing is to have the homebrew at /usr/local/bin, so it can be translated by rosetta2.

Roberta

**Comment by npcarter** [comment]
Great, glad to hear it's working! I wish I knew why things are behaving differently for you than for me, but I don't know of a good way to figure that out.

-﻿Nick

[Nick Carter - Chat @ Spike](https://spikenow.com/r/a/?ref=spike-organic-signature&_ts=xp5rb)	[xp5rb]

On March 6, 2021 at 8:52 GMT, rob2go <notifications@github.com> wrote:

ok thank you for your help.
I installed kityy. I verified on the Activity monitor that it was effectively running "bash" under intel. Everything fine but even like that, not possible to install hmmer.

So I installed homebrew under /usr/local/bin
And now it works :-)

arch -x86_64 /usr/local/bin/brew install hmmer

Like that, does not matter if I install using kitty or the apple terminal. The important thing is to have the homebrew at /usr/local/bin, so it can be translated by rosetta2.

Roberta
